subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/ErrorHandlingModuleComponentRepositoryTest.groovy

```diff
@@ -44,7 +47,23 @@ class ErrorHandlingModuleComponentRepositoryTest extends Specification {
     private static final String REPOSITORY_ID = 'abc'
     def delegate = Mock(ModuleComponentRepositoryAccess)
     def repositoryBlacklister = Mock(RepositoryBlacklister)
-    def someException = new RuntimeException('Something went wrong')
+
+    @Shared
+    def runtimeError = new RuntimeException('Something went wrong')
+    @Shared
+    def connectTimeout = new SocketTimeoutException()
+    @Shared
+    def cannotConnect = new HttpHostConnectException(null, null)
+    @Shared
+    def missing = status(404)
+    @Shared
+    def serverError = status(500)
```

Comment:
Would be great to have a `500..599` range here, right now the code will work but test doesn't guard against potential future changes which may stop retrying say `504`

Reply:
done!